libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1887,20 +1989,39 @@ string YulUtilFunctions::negateNumberCheckedFunction(Type const& _type)
 string YulUtilFunctions::zeroValueFunction(Type const& _type)
 {
 	solUnimplementedAssert(_type.sizeOnStack() == 1, "Stacksize not yet implemented!");
-	solUnimplementedAssert(_type.isValueType(), "Zero value for non-value types not yet implemented");
+	solAssert(_type.category() != Type::Category::Mapping, "");
 
 	string const functionName = "zero_value_for_" + _type.identifier();
 
 	return m_functionCollector.createFunction(functionName, [&]() {
-		return Whiskers(R"(
+		Whiskers templ(R"(
 			function <functionName>() -> ret {
-				<body>
+				ret := <zeroValue>
 			}
-		)")
-			("functionName", functionName)
-			("body", "ret := 0x0")
-			.render();
-		});
+		)");
+		templ("functionName", functionName);
+
+		if (_type.isValueType())
+			templ("zeroValue", "0");
```

Comment:
Maybe add some more assertions here? `hasSimpleZeroValueinMemory() || is internal or external function type`.

Reply:
Done